### PR TITLE
Added clearing of setTimeout ID so that it can be cleared when the component is unmounted

### PR DESCRIPTION
**Pull request by rj-david**
This avoids this bug:
https://github.com/schiehll/react-alert/issues/79

With this fix, the setTimeout() function is cleared whenever the component is unmounted. This also supports multiple alerts.

Please disregard the changes in .gitignore. The fix only involved src/Provider.js

### CODE REVIEW

**Comment by schiehll** [comment]
Hey, @rj-david! Thank you for the PR!

I will review your changes so we can merge it.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,4 @@
 /node_modules
-dist
```
Comment (schiehll) [inline .gitignore]: You will need to add `dist` back into the file, so we don't have to move the `dist` folder around.

File: package.json [hunk 2]
```diff
@@ -64,5 +64,8 @@
       "src/*.js",
       "!src/index.js"
     ]
+  },
+  "dependencies": {
+    "babel-core": "^6.26.3"
```
Comment (schiehll) [inline package.json]: We don't need this dependency. Please remove it. 

If you need `babel-core` to build and test locally you can add it to the `devDependencies`.

File: src/Provider.js [hunk 3]
```diff
@@ -61,9 +63,13 @@ class Provider extends Component {
     alert.close = () => this.remove(alert)
 
     if (alert.options.timeout) {
-      setTimeout(() => {
+      let timerId = setTimeout(() => {
```
Comment (schiehll) [inline src/Provider.js]: We don't need to mutate the `timerId` id later, so we can declare it as a `const`, like `const timerId = `.

File: src/Provider.js [hunk 4]
```diff
@@ -114,6 +120,9 @@ class Provider extends Component {
   }
 
   componentWillUnmount() {
+    if (this.timerId.length) {
```
Comment (schiehll) [inline src/Provider.js]: There's no need for this check here, the `forEach` will not be executed if the array is empty.

Also, if you just want to execute a function for each element you can simply pass that function as the parameter.

So we can simplify this 3 lines to just `this.timerId.forEach(clearTimeout)`.

File: .gitignore (hunk 1 above)
Comment (rj-david) [reply .gitignore]: Added back

File: package.json (hunk 2 above)
Comment (rj-david) [reply package.json]: Removed dependency

File: src/Provider.js (hunk 3 above)
Comment (rj-david) [reply src/Provider.js]: Changed to const

File: src/Provider.js (hunk 4 above)
Comment (rj-david) [reply src/Provider.js]: Done with the recommendation

**Pull request merged**